From 48e328efaa5234f45a10103fc2a0a8ed2987b947 Mon Sep 17 00:00:00 2001
From: Croydon <cr0ydon@outlook.com>
Date: Wed, 5 Sep 2018 08:40:49 +0200
Subject: [PATCH] Name the library always freeglut and not freeglut_static

This causes trouble to find the library afterwards when built static. Inspired by https://github.com/Microsoft/vcpkg/blob/68bfea163aaa59d59e9e45954fa1777028c7f770/ports/freeglut/portfile.cmake
---
 CMakeLists.txt            | 14 +++++++-------
 include/GL/freeglut_std.h |  5 ++---
 2 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3a06b2d..3550148 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -351,7 +351,7 @@ IF(FREEGLUT_BUILD_SHARED_LIBS)
     ADD_LIBRARY(freeglut SHARED ${FREEGLUT_SRCS})
 ENDIF()
 IF(FREEGLUT_BUILD_STATIC_LIBS)
-    ADD_LIBRARY(freeglut_static STATIC ${FREEGLUT_SRCS})
+    ADD_LIBRARY(freeglut STATIC ${FREEGLUT_SRCS})
 ENDIF()


@@ -361,12 +361,12 @@ IF(WIN32)
         SET_TARGET_PROPERTIES(freeglut PROPERTIES COMPILE_FLAGS -DFREEGLUT_EXPORTS)
     ENDIF()
     IF(FREEGLUT_BUILD_STATIC_LIBS)
-        SET_TARGET_PROPERTIES(freeglut_static PROPERTIES COMPILE_FLAGS -DFREEGLUT_STATIC)
+        SET_TARGET_PROPERTIES(freeglut PROPERTIES COMPILE_FLAGS -DFREEGLUT_STATIC)
         # need to set machine:x64 for linker, at least for VC10, and
         # doesn't hurt for older compilers:
         # http://public.kitware.com/Bug/view.php?id=11240#c22768
         IF (CMAKE_CL_64)
-            SET_TARGET_PROPERTIES(freeglut_static PROPERTIES STATIC_LIBRARY_FLAGS "/machine:x64")
+            SET_TARGET_PROPERTIES(freeglut PROPERTIES STATIC_LIBRARY_FLAGS "/machine:x64")
         ENDIF()
     ENDIF()
 ELSE()
@@ -399,7 +399,7 @@ ELSE()
       SET_TARGET_PROPERTIES(freeglut PROPERTIES VERSION ${SO_MAJOR}.${SO_MINOR}.${SO_REV} SOVERSION ${SO_MAJOR} OUTPUT_NAME ${LIBNAME})
     ENDIF()
     IF(FREEGLUT_BUILD_STATIC_LIBS)
-      SET_TARGET_PROPERTIES(freeglut_static PROPERTIES OUTPUT_NAME ${LIBNAME})
+      SET_TARGET_PROPERTIES(freeglut PROPERTIES OUTPUT_NAME ${LIBNAME})
     ENDIF()
     IF(ANDROID)
         # Not in CMake toolchain file, because the toolchain
@@ -417,7 +417,7 @@ IF(FREEGLUT_BUILD_SHARED_LIBS)
     TARGET_LINK_LIBRARIES(freeglut ${LIBS})
 ENDIF()
 IF(FREEGLUT_BUILD_STATIC_LIBS)
-    TARGET_LINK_LIBRARIES(freeglut_static ${LIBS})
+    TARGET_LINK_LIBRARIES(freeglut ${LIBS})
 ENDIF()

 IF(FREEGLUT_BUILD_SHARED_LIBS)
@@ -433,7 +433,7 @@ IF(FREEGLUT_BUILD_SHARED_LIBS)
     ENDIF()
 ENDIF()
 IF(FREEGLUT_BUILD_STATIC_LIBS)
-    INSTALL(TARGETS freeglut_static
+    INSTALL(TARGETS freeglut
             RUNTIME DESTINATION bin
             LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
@@ -463,7 +463,7 @@ MACRO(ADD_DEMO name)
         ENDIF()
         IF(FREEGLUT_BUILD_STATIC_LIBS)
             ADD_EXECUTABLE(${name}_static ${ARGN})
-            TARGET_LINK_LIBRARIES(${name}_static ${DEMO_LIBS} freeglut_static)
+            TARGET_LINK_LIBRARIES(${name}_static ${DEMO_LIBS} freeglut)
             SET_TARGET_PROPERTIES(${name}_static PROPERTIES COMPILE_FLAGS -DFREEGLUT_STATIC)
             IF(WIN32 AND MSVC)
                 SET_TARGET_PROPERTIES(${name}_static PROPERTIES DEBUG_POSTFIX d)
diff --git a/include/GL/freeglut_std.h b/include/GL/freeglut_std.h
index a658c7c..051187e 100644
--- a/include/GL/freeglut_std.h
+++ b/include/GL/freeglut_std.h
@@ -71,9 +71,9 @@
         /* Link with Win32 static freeglut lib */
 #       if FREEGLUT_LIB_PRAGMAS
 #           ifdef NDEBUG
-#              pragma comment (lib, "freeglut_static.lib")
+#              pragma comment (lib, "freeglut.lib")
 #           else
-#              pragma comment (lib, "freeglut_staticd.lib")
+#              pragma comment (lib, "freeglut.lib")
 #           endif
 #       endif

@@ -650,4 +650,3 @@ static int FGAPIENTRY FGUNUSED glutCreateMenu_ATEXIT_HACK(void (* func)(int)) {
 /*** END OF FILE ***/

 #endif /* __FREEGLUT_STD_H__ */
-
--
